components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2450,6 +2492,196 @@ public void updateGroup(Group oldGroup, Group newGroup) throws CharonException {
+                    for (Map<String, String> memberObject : memberOperationValues) {
+                        prepareAddedRemovedMemberLists(addedMembers, deletedMembers, newlyAddedMemberIds,
+                                memberOperation, memberObject);
+                    }
+                }
+            }
+
+            String userStoreDomainForGroup = IdentityUtil.extractDomainFromName(newGroupName);
+            Set<String> temporaryMembers = new HashSet<>();
+
+            if (isNotInternalOrApplicationGroup(userStoreDomainForGroup) && (!addedMembers.isEmpty()
+                    || !deletedMembers.isEmpty())) {
+                for (String member : addedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                addedMembers.clear();
+                addedMembers.addAll(temporaryMembers);
+                temporaryMembers.clear();
+
+                for (String member : deletedMembers) {
+                    String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member),
+                            userStoreDomainForGroup);
+                    temporaryMembers.add(username);
+                }
+
+                deletedMembers.clear();
+                deletedMembers.addAll(temporaryMembers);
+            }
+
+            // Check for deleted members
+            Set<String> deletedMemberIdsFromUserstore =
+                    getMemberValuesFromUserstore(deletedMembers, userStoreDomainForGroup, newGroupName);
+
+            // Check for added members
+            Set<String> addedMemberIdsFromUserstore =
+                    getMemberValuesFromUserstore(addedMembers, userStoreDomainForGroup, newGroupName);
+
+            // Validate the memberIds sent in the update request against the Ids retrieved from the user store.
+            if (isNotEmpty(addedMembers)) {
+                validateUserIds(addedMemberIdsFromUserstore, newlyAddedMemberIds);
+            }
+
+            /*
+            Set thread local property to signal the downstream SCIMUserOperationListener
+            about the provisioning route.
+            */
+            SCIMCommonUtils.setThreadLocalIsManagedThroughSCIMEP(true);
+
+             /*
+             We do not update Identity_SCIM DB here since it is updated in SCIMUserOperationListener's methods.
+             Update the group with added members and deleted members.
+              */
+            if (isNotEmpty(addedMembers) || isNotEmpty(deletedMembers)) {
+                carbonUM.updateUserListOfRoleWithID(newGroupName,
+                        deletedMemberIdsFromUserstore.toArray(new String[0]),
+                        addedMemberIdsFromUserstore.toArray(new String[0]));
+            }
+
+        } catch (UserStoreException e) {
+            log.error(e);
```

Comment:
Do we need to log and throw the error at the same time?

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/280/commits/36add81b74fe1bfc479adf2b8217e00af089b8f7